hedera-node/src/test/java/com/hedera/services/txns/contract/ContractDeleteTransitionLogicTest.java

```diff
@@ -97,109 +105,66 @@ void hasCorrectApplicability() {
 
-		// then:
-		verify(txnCtx).setStatus(ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
-		verifyNoInteractions(delegate);
+		thenWillFailWith(ACCOUNT_EXPIRED_AND_PENDING_REMOVAL);
 	}
 
 	@Test
-	void capturesBadDelete() {
-		// setup:
-		TransactionRecord deleteRec = TransactionRecord.newBuilder()
-				.setReceipt(TransactionReceipt.newBuilder()
-						.setStatus(MODIFYING_IMMUTABLE_CONTRACT)
-						.build())
-				.build();
-
+	void rejectsInvalidCid() {
 		givenValidTxnCtx();
-		// and:
-		given(delegate.perform(contractDeleteTxn, consensusTime)).willReturn(deleteRec);
+		given(accountStore.loadContract(Id.fromGrpcAccount(asAccount(TARGET)))).willThrow(new InvalidTransactionException(CONTRACT_DELETED));
 
-		// when:
-		subject.doStateTransition();
-
-		// then:
-		verify(txnCtx).setStatus(MODIFYING_IMMUTABLE_CONTRACT);
+		thenWillFailWith(CONTRACT_DELETED);
 	}
 
-	@Test
-	void followsHappyPathWithOverrides() {
-		// setup:
-		TransactionRecord updateRec = TransactionRecord.newBuilder()
-				.setReceipt(TransactionReceipt.newBuilder()
-						.setStatus(SUCCESS)
-						.build())
-				.build();
-
-		givenValidTxnCtx();
-		// and:
-		given(delegate.perform(contractDeleteTxn, consensusTime)).willReturn(updateRec);
-
-		// when:
-		subject.doStateTransition();
-
-		// then:
-		verify(txnCtx).setStatus(SUCCESS);
+	private void thenWillFailWith(ResponseCodeEnum code) {
+		assertFailsWith(() -> subject.doStateTransition(), code);
+		verify(target, never()).delete();
+		verify(accountStore, never()).persistAccount(beneficiary);
+		verify(accountStore, never()).persistAccount(target);
 	}
 
-	@Test
-	void acceptsOkSyntax() {
-		givenValidTxnCtx();
-
-		// expect:
-		assertEquals(OK, subject.semanticCheck().apply(contractDeleteTxn));
+	private void givenLoadedAccount() {
+		given(accountStore.loadBeneficiary(Id.fromGrpcAccount(BENEFICIARY))).willReturn(beneficiary);
 	}
 
-	@Test
-	void rejectsInvalidCid() {
-		givenValidTxnCtx();
-		// and:
-		given(validator.queryableContractStatus(target, contracts)).willReturn(CONTRACT_DELETED);
-
-		// expect:
-		assertEquals(CONTRACT_DELETED, subject.semanticCheck().apply(contractDeleteTxn));
+	private void givenLoadedContract() {
+		given(accountStore.loadContract(Id.fromGrpcAccount(asAccount(TARGET)))).willReturn(target);
 	}
 
-	@Test
-	void translatesUnknownException() {
-		givenValidTxnCtx();
-
-		given(delegate.perform(any(), any())).willThrow(IllegalStateException.class);
-
-		// when:
-		subject.doStateTransition();
-
-		// then:
-		verify(txnCtx).setStatus(FAIL_INVALID);
+	private void givenValidTxnCtx() {
+		var op = TransactionBody.newBuilder()
+				.setTransactionID(ourTxnId())
+				.setContractDeleteInstance(
+						ContractDeleteTransactionBody.newBuilder()
+								.setTransferAccountID(BENEFICIARY)
+								.setContractID(TARGET));
+		contractDeleteTxn = op.build();
+		lenient().when(accessor.getTxn()).thenReturn(contractDeleteTxn);
+		lenient().when(txnCtx.accessor()).thenReturn(accessor);
```

Comment:
The usage of `lenient()` can be removed by mocking all the needed behaviour in the exact  test case where we need it.
Test cases will become a little bit larger, but you can later extract the common parts in a `@BeforeEach` .
Please note that mockito (when using `@ExtendWith` ) will throw an exception every time you are "overmocking" some behaviour.